lib/Logger/LogThread.cpp

```diff
@@ -75,21 +75,32 @@ void LogThread::wakeup() noexcept {
 bool LogThread::hasMessages() const noexcept { return !_messages.empty(); }
 
 void LogThread::run() {
+  constexpr uint64_t initialWaitTime = 25 * 1000;
+
+  uint64_t waitTime = initialWaitTime;
   while (!isStopping() && Logger::_active.load()) {
-    processPendingMessages();
+    bool worked = processPendingMessages();
+    if (worked) {
+      waitTime = initialWaitTime;
+    } else {
+      waitTime *= 2;
+      waitTime = std::min(initialWaitTime, waitTime);
```

Comment:
Won't this always just be `initialWaitTime` then?

Reply:
True. Thanks for pointing this out. I am notoriously bad at using min/max :smile: